Can't cmake on Ubuntu 18.04

Can't build latest master. I'm running Linux 18.04. 

Versions:
- X++ latest master
- **cmake --version** `cmake version 3.10.2`
- **gcc -v**:
`Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/7/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu 7.4.0-1ubuntu1~18.04.1' --with-bugurl=file:///usr/share/doc/gcc-7/README.Bugs --enable-languages=c,ada,c++,go,brig,d,fortran,objc,obj-c++ --prefix=/usr --with-gcc-major-version-only --program-suffix=-7 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --with-sysroot=/ --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-libmpx --enable-plugin --enable-default-pie --with-system-zlib --with-target-system-zlib --enable-objc-gc=auto --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
gcc version 7.4.0 (Ubuntu 7.4.0-1ubuntu1~18.04.1) `

I'm attaching CMake files:
[CMakeError.log](https://github.com/xournalpp/xournalpp/files/3744500/CMakeError.log)
[CMakeOutput.log](https://github.com/xournalpp/xournalpp/files/3744501/CMakeOutput.log)

**Output when running cmake:**
`CMake Error at cmake/find/FindCXX17.cmake:72 (target_compile_features):
  Cannot specify compile features for imported target "cxx17".
Call Stack (most recent call first):
  CMakeLists.txt:64 (find_package)


-- Checking for module 'gtk-mac-integration'
--   No package 'gtk-mac-integration' found
Enable Xournal++ Plugins
Installing desktop files

Configuration:
	Compiler:                   /usr/bin/c++
	CppUnit enabled:            OFF

-- Configuring incomplete, errors occurred!
See also "/home/mark/Documents/Repos/External/xournalpp/build/CMakeFiles/CMakeOutput.log".
See also "/home/mark/Documents/Repos/External/xournalpp/build/CMakeFiles/CMakeError.log".`

Update -- forcing use of gcc-8 and g++-8 doesn't work. 

Did an apt install g++-8 and gcc-8. 

Forcing cmake to use it -- `cmake -DCMAKE_C_COMPILER=/usr/bin/gcc-8 -DCMAKE_CXX_COMPILER=/usr/bin/g++-8 ..`

Cmake output files:
[CMakeError.log](https://github.com/xournalpp/xournalpp/files/3744993/CMakeError.log)
[CMakeOutput.log](https://github.com/xournalpp/xournalpp/files/3744994/CMakeOutput.log)


Output to shell:
-- The CXX compiler identification is GNU 8.3.0
-- The C compiler identification is GNU 8.3.0
-- Check for working CXX compiler: /usr/bin/g++-8
-- Check for working CXX compiler: /usr/bin/g++-8 -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Check for working C compiler: /usr/bin/gcc-8
-- Check for working C compiler: /usr/bin/gcc-8 -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Found Git: /usr/bin/git (found version "2.17.1") 
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.1") 
-- Looking for C++ include optional
-- Looking for C++ include optional - found
CMake Error at cmake/find/FindCXX17.cmake:72 (target_compile_features):
  Cannot specify compile features for imported target "cxx17".
Call Stack (most recent call first):
  CMakeLists.txt:64 (find_package)


-- Looking for backtrace
-- Looking for backtrace - found
-- backtrace facility detected in default set of libraries
-- Found Backtrace: /usr/include  
-- Checking for module 'glib-2.0 >= 2.32.0'
--   Found glib-2.0 , version 2.56.4
-- Checking for module 'gtk+-3.0 >= 3.18.9'
--   Found gtk+-3.0 , version 3.22.30
-- Checking for module 'gthread-2.0 >= 2.4.0'
--   Found gthread-2.0 , version 2.56.4
-- Checking for module 'libxml-2.0 >= 2.0.0'
--   Found libxml-2.0 , version 2.9.4
-- Checking for module 'poppler-glib >= 0.41.0'
--   Found poppler-glib , version 0.62.0
-- Found ZLIB: /usr/lib/x86_64-linux-gnu/libz.so (found version "1.2.11") 
-- Checking for module 'libzip >= 1.0.1'
--   Found libzip , version 1.1.2
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Checking for module 'portaudiocpp >= 12'
--   Found portaudiocpp , version 12
-- Checking for module 'sndfile >= 1.0.25'
--   Found sndfile , version 1.0.28
-- Checking for module 'gtk-mac-integration'
--   No package 'gtk-mac-integration' found
-- Checking for module 'lua5.3'
--   Found lua5.3, version 5.3.3
Enable Xournal++ Plugins
-- Found Gettext: /usr/bin/xgettext (found version "0.19.8.1") 
Installing desktop files

Configuration:
	Compiler:                   /usr/bin/g++-8
	CppUnit enabled:            OFF

-- Configuring incomplete, errors occurred!
See also "/home/mark/Documents/Repos/External/xournalpp/build/CMakeFiles/CMakeOutput.log".
See also "/home/mark/Documents/Repos/External/xournalpp/build/CMakeFiles/CMakeError.log".

@Technius 
I just tried `cmake -DCMAKE_C_COMPILER=/usr/bin/clang -DCMAKE_CXX_COMPILER=/usr/bin/clang++ ..`, but seem to get same error.

[CMakeError.log](https://github.com/xournalpp/xournalpp/files/3745012/CMakeError.log)
[CMakeOutput.log](https://github.com/xournalpp/xournalpp/files/3745014/CMakeOutput.log)

*shell output*
-- The CXX compiler identification is Clang 6.0.0
-- The C compiler identification is Clang 6.0.0
-- Check for working CXX compiler: /usr/bin/clang++
-- Check for working CXX compiler: /usr/bin/clang++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Check for working C compiler: /usr/bin/clang
-- Check for working C compiler: /usr/bin/clang -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Found Git: /usr/bin/git (found version "2.17.1") 
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.1") 
-- Looking for C++ include optional
-- Looking for C++ include optional - found
CMake Error at cmake/find/FindCXX17.cmake:72 (target_compile_features):
  Cannot specify compile features for imported target "cxx17".
Call Stack (most recent call first):
  CMakeLists.txt:64 (find_package)


-- Looking for backtrace
-- Looking for backtrace - found
-- backtrace facility detected in default set of libraries
-- Found Backtrace: /usr/include  
-- Checking for module 'glib-2.0 >= 2.32.0'
--   Found glib-2.0 , version 2.56.4
-- Checking for module 'gtk+-3.0 >= 3.18.9'
--   Found gtk+-3.0 , version 3.22.30
-- Checking for module 'gthread-2.0 >= 2.4.0'
--   Found gthread-2.0 , version 2.56.4
-- Checking for module 'libxml-2.0 >= 2.0.0'
--   Found libxml-2.0 , version 2.9.4
-- Checking for module 'poppler-glib >= 0.41.0'
--   Found poppler-glib , version 0.62.0
-- Found ZLIB: /usr/lib/x86_64-linux-gnu/libz.so (found version "1.2.11") 
-- Checking for module 'libzip >= 1.0.1'
--   Found libzip , version 1.1.2
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Checking for module 'portaudiocpp >= 12'
--   Found portaudiocpp , version 12
-- Checking for module 'sndfile >= 1.0.25'
--   Found sndfile , version 1.0.28
-- Checking for module 'gtk-mac-integration'
--   No package 'gtk-mac-integration' found
-- Checking for module 'lua5.3'
--   Found lua5.3, version 5.3.3
Enable Xournal++ Plugins
-- Found Gettext: /usr/bin/xgettext (found version "0.19.8.1") 
Installing desktop files

Configuration:
	Compiler:                   /usr/bin/clang++
	CppUnit enabled:            OFF

-- Configuring incomplete, errors occurred!
See also "/home/mark/Documents/Repos/External/xournalpp/build/CMakeFiles/CMakeOutput.log".
See also "/home/mark/Documents/Repos/External/xournalpp/build/CMakeFiles/CMakeError.log".

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -61,9 +61,6 @@ macro (add_includes_ldflags LDFLAGS INCLUDES)
     set (xournalpp_INCLUDE_DIRS ${xournalpp_INCLUDE_DIRS} "${INCLUDES}")
 endmacro (add_includes_ldflags LDFLAGS INCLUDES)
 
-find_package(CXX17 REQUIRED COMPONENTS optional)
-set(xournalpp_LDFLAGS ${xournalpp_LDFLAGS} cxx17)
-
 # libexec
 if (WIN32)
     set (xournalpp_LDFLAGS ${xournalpp_LDFLAGS} "-mwindows")

--- a/cmake/find/FindCXX17.cmake
+++ b/cmake/find/FindCXX17.cmake
@@ -23,13 +23,9 @@ set(CMAKE_CXX_STANDARD_REQUIRED on)
 
 # Normalize and check the component list we were given
 set(want_components ${CXX17_FIND_COMPONENTS})
-if (CXX17_FIND_COMPONENTS STREQUAL "")
-#Nothing to do here
-endif ()
 
 # Warn on any unrecognized components
-if (NOT extra_components STREQUAL "")
-    set(extra_components ${want_components})
+if (extra_components)
     list(REMOVE_ITEM extra_components optional filesystem map)
     foreach (component IN LISTS extra_components)
         message(WARNING "Extraneous find_package component for Filesystem: ${component}")
@@ -63,13 +59,6 @@ if ("map" IN_LIST want_components)
     endif ()
 endif ()
 
-if (TARGET cxx17)
-    # This module has already been processed. Don't do it again.
-    return()
-endif ()
-
-add_library(cxx17 INTERFACE IMPORTED)
-target_compile_features(cxx17 INTERFACE cxx_std_17)
 set(_found TRUE)
 
 cmake_pop_check_state()

--- a/src/util/CMakeLists.txt
+++ b/src/util/CMakeLists.txt
@@ -3,6 +3,4 @@ unset (util_SOURCES)
 file (GLOB_RECURSE util_SOURCES *.cpp)
 
 add_library (util STATIC ${util_SOURCES})
-find_package(CXX17 REQUIRED)
-target_link_libraries(util PRIVATE cxx17)
 target_compile_features(util PUBLIC ${PROJECT_CXX_FEATURES})

